agr_literature_service/api/crud/search_crud.py

```diff
@@ -10,13 +12,90 @@
 
 from fastapi import HTTPException, status
 
+logger = logging.getLogger(__name__)
+
+
+def date_str_to_micro_seconds(date_str: str, start_of_day: bool):
+    # convert string to Datetime int that is stored in Elastic search
+    # initial strings are in the format:- "2010-10-28T04:00:00.000"
+    # So just grab chars before T and converts to seconds after epoch
+    # then mulitply by 1000000 and convert to int.
+    date_time = datetime.strptime(date_str.split('T')[0], '%Y-%m-%d')
+    if start_of_day:
+        date_time = date_time.replace(hour=0, minute=0, second=0, microsecond=0)
+    else:
+        date_time = date_time.replace(hour=23, minute=59, second=59, microsecond=0)
+    return int(date_time.timestamp() * 1000000)
+
+
+def search_date_range(es_body,
+                      date_pubmed_modified: Optional[List[str]] = None,
+                      date_pubmed_arrive: Optional[List[str]] = None,
+                      date_published: Optional[List[str]] = None):
+    # date_pubmed_X is split to just get the date and remove the time element
+    # as elastic search does a tr comparison and if the end date has the time
+    # element stil in it fails even if the date bits match as the string is longer.
+    if "must" not in es_body["query"]["bool"]["filter"]["bool"]:
+        es_body["query"]["bool"]["filter"]["bool"]["must"] = []
+    if date_pubmed_modified:
+        es_body["query"]["bool"]["filter"]["bool"]["must"].append(
+            {
+                "range": {
+                    "date_last_modified_in_pubmed": {
+                        "gte": date_pubmed_modified[0].split('T')[0],
```

Comment:
Is this necessary?  I didn't think these dates had a time component?

Reply:
The values in the database do not have time component but the ui returns a string with the time components which messes things up in the search as one string is longer/bigger than the other if the dates match.